modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2558,6 +2559,11 @@ def concat(
             left_parts = self._partitions
             right_parts = [o._partitions for o in others]
             new_lengths = self._row_lengths_cache
+            other_widths = [width for o in others for width in o._column_widths_cache]
```

Comment:
@anmyachev could you please add a comment like [the one above](https://github.com/modin-project/modin/pull/4728/files#r931572472) explaining that we can only do this for `COL_WISE` because `concat` might rebalance partitions for `ROW_WISE`?